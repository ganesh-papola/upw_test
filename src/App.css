

p,
h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0 !important;
}

input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

/* Firefox */
input[type="number"] {
  -moz-appearance: textfield;
}
