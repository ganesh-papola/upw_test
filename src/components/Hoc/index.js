export { default as LoaderHoc } from "./LoaderHoc";
