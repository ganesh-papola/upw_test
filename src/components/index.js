export * from "./common";
export * from "./Hoc";
