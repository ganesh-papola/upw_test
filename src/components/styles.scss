@import "../shared/styles/index.scss";

$max-zindex: 1000;
$mid-zindex: 999;
