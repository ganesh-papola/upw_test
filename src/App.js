import "./App.css";
import { Provider } from "react-redux";
import { persistor, store } from "./redux/store";
import Routes from "./routes/routes";
import { PersistGate } from "redux-persist/integration/react";
import { HelmetProvider } from "react-helmet-async";
import { LoaderHoc } from "./components";

function App() {
  return (
    <Provider store={store}>
      <PersistGate persistor={persistor}>
        <HelmetProvider>
          <LoaderHoc>
            <Routes />
          </LoaderHoc>
        </HelmetProvider>
      </PersistGate>
    </Provider>
  );
}

export default App;
