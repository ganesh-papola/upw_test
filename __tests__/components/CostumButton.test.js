import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { CustomButton } from "../../src/components"

describe('CustomButton test', () => {
    const props = {
        title : "Test Button",
        type : "button",
        onClick : jest.fn(),
        buttonClass : "",
    }
    it('this is components snapshot rendering test', () => {
        const { container } = render(
          <CustomButton {...props} />
        );
        expect(container).toMatchSnapshot();
      });
      it('should add click the button', () => {
        const { getByText } = render(<CustomButton {...props} />)
        const click = getByText("Test Button")
        fireEvent.click(click)
        expect(props.onClick).toHaveBeenCalledTimes(1)
    })
})
