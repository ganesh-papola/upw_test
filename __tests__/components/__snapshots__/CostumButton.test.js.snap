// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton test this is components snapshot rendering test 1`] = `
<div>
  <div
    class="buttonContainer"
  >
    <button
      class="button"
      type="button"
    >
      Test Button
    </button>
  </div>
</div>
`;
